test/simulator/moves/trickroom.js

```diff
@@ -0,0 +1,116 @@
+	});
+
+	it('should cause slower Pokemon to move before faster Pokemon in a priority bracket', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Bronzong', ability: 'heatproof', moves: ['spore', 'trickroom']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Ninjask', ability: 'speedboost', moves: ['poisonjab', 'spore']}]);
+		battle.choose('p1', 'move 2');
+		battle.commitDecisions();
+		battle.choose('p2', 'move 2');
+		battle.commitDecisions();
+		assert.strictEqual(battle.p1.active[0].status, '');
+		assert.strictEqual(battle.p2.active[0].status, 'slp');
+	});
+
+	it('should not allow Pokemon using a lower priority move to act before other Pokemon', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Bronzong', ability: 'heatproof', moves: ['spore', 'trickroom']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Ninjask', ability: 'speedboost', moves: ['poisonjab', 'protect']}]);
+		battle.choose('p1', 'move 2');
+		battle.commitDecisions();
+		battle.choose('p2', 'move 2');
+		battle.commitDecisions();
+		assert.strictEqual(battle.p1.active[0].status, '');
+		assert.strictEqual(battle.p2.active[0].status, '');
+	});
+
+	it('should also affect the activation order for abilities and other non-move actions', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [
+			{species: 'Bronzong', ability: 'heatproof', moves: ['trickroom', 'explosion']},
+			{species: 'Hippowdon', ability: 'sandstream', moves: ['protect']}
+		]);
+		battle.join('p2', 'Guest 2', 1, [
+			{species: 'Ninjask', ability: 'speedboost', moves: ['shellsmash']},
+			{species: 'Ninetales', ability: 'drought', moves: ['protect']}
+		]);
+		battle.commitDecisions();
+		battle.choose('p1', 'move 2');
+		battle.commitDecisions();
+		battle.choose('p1', 'switch 2');
+		battle.choose('p2', 'switch 2');
+		battle.commitDecisions();
+		assert.strictEqual(battle.p1.active[0].speciesid, 'hippowdon');
+		assert.strictEqual(battle.p2.active[0].speciesid, 'ninetales');
+		assert.strictEqual(battle.effectiveWeather(), 'sunnyday');
+	});
+
+	it('should roll over and cause Pokemon with 1809 or more speed to outspeed Pokemon with 1808 or less', function () {
+		battle = BattleEngine.Battle.construct('battle-trickroom-rollover', 'customgame');
+		battle.join('p1', 'Guest 1', 1, [{species: 'Ninjask', ability: 'swarm', moves: ['protect', 'spore']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Ninjask', ability: 'blaze', moves: ['spore', 'trickroom']}]);
+		battle.commitDecisions(); // Team Preview
+		battle.choose('p2', 'move 2');
+		battle.commitDecisions();
+
+		battle.on('ModifySpe', battle.getFormat(), function (spe, pokemon) {
+			// We are using abilities to decide who has what speed, exactly.
+			// In-game, this can be tested (albeit slightly less precicely) with Choice Scarfed Ninjasks with various speed boosts and EVs
```

Comment:
I'd suggest transcripting the Ninjask test here.
Ideally, all our battle tests should be reproducible as-is in cartridge.